synapse/federation/federation_server.py

```diff
@@ -469,7 +470,12 @@ async def process_pdus_for_room(room_id: str) -> None:
                     )
                     for pdu in pdus_by_room[room_id]:
                         event_id = pdu.event_id
-                        pdu_results[event_id] = e.error_dict()
+                        if isinstance(e, UnstableSpecAuthError):
```

Comment:
Bleurgh, I'm not a fan of using `isinstance` TBH, its a bit of a smell. I'd rather we changed the signature of `error_dict` across all the errors to avoid this. At which point you probably do want to pass in either the config or HS object to make the `error_dict` API sane.

Reply:
>Bleurgh, I'm not a fan of using isinstance TBH, its a bit of a smell. I'd rather we changed the signature of error_dict across all the errors to avoid this.

Happy to :+1: 

>At which point you probably do want to pass in either the config or HS object to make the error_dict API sane.

I'll try it with the config. I *could* see that coming up again. It might just be prior experience but I have a natural aversion to passing in chunky classes into smaller classes.